modules/prometheus/client/src/test/scala/higherkindness/mu/rpc/prometheus/client/BaseMonitorClientInterceptorTests.scala

```diff
@@ -327,12 +325,13 @@ abstract class BaseMonitorClientInterceptorTests extends RpcBaseTestSuite {
       import clientRuntime._
 
       (for {
-        _         <- serverStart[ConcurrentMonad]
-        _         <- unary[ConcurrentMonad]
-        _         <- clientStreaming[ConcurrentMonad]
-        assertion <- check
-        _         <- serverStop[ConcurrentMonad]
-      } yield assertion).unsafeRunSync()
+        _ <- serverStart[ConcurrentMonad]
+        _ <- unary[ConcurrentMonad]
+        _ <- clientStreaming[ConcurrentMonad]
+        _ <- serverStop[ConcurrentMonad]
+      } yield ()).unsafeRunSync()
+
+      check.unsafeRunSync()
```

Comment:
These changes seem weird to me 🤔 , are we doing the check at the end? Is that correct?

Reply:
This is a fix to prevent random failures. Basically this is what I explained here: https://github.com/higherkindness/mu/pull/564

I'm not sure if after merging master this is still necessary. I'll check it out.

Reply:
It's still failing:

```bash
[info] - should work when combining multiple calls *** FAILED ***
[info]   1 was not equal to 
[info]   2 (BaseMonitorClientInterceptorTests.scala:321)
```

It seems we still need the fix